### DISCUSSION TITLE
feat: Track Exceptions while using Decorators

### DISCUSSION THREAD

**Discussion by Manonandan** [langfuse/langfuse#2529] [Ideas]
### Describe the feature or potential improvement

The `@observe` decorator captures the input and output of the method in the trace/span. 
Similarly, it is great to have the exceptions captured.

**Comment by hassiebp** [comment]
Hi @Manonandan - thanks for your suggestion! The decorator implementation should already handle raised exceptions inside a decorated function. See the related [test case here](https://github.com/langfuse/langfuse-python/blob/5ef7f75dbb41af04e9e0f2a41b42cad4326f7ff8/tests/test_decorators.py#L106). Have you noticed unexpected behavior with respect to this?
